Controllers/PostController.cs

```diff
@@ -189,4 +189,28 @@ public IActionResult DeletePost(int id)
 
         return NoContent();
     }
+
+    [HttpGet("user/{id}")]
+    [Authorize]
+    public IActionResult GetPostsByUserId(int id)
+    {
+        List <Post> postsByUser = _dbContext
+            .Posts.Include(p => p.Author)
+            .Include(p => p.Comments)
+            .ThenInclude(c => c.Commenteer)
+            .ThenInclude(u => u.IdentityUser)
+            .Include(p => p.PostTags)
+            .Where(p => p.Id == id).ToList();
```

Comment:
This will filter posts by the posts's Id, which will only return one post and it may not be the author's post. I think you meant to filter by authorId

Reply:
Good Catch!